scaladoc/src/dotty/tools/scaladoc/tasty/SymOps.scala

```diff
@@ -146,11 +146,19 @@ object SymOps:
 
     def extendedSymbol: Option[reflect.ValDef] =
       import reflect.*
-      Option.when(sym.isExtensionMethod){
-        val termParamss = sym.tree.asInstanceOf[DefDef].termParamss
-        if sym.isLeftAssoc || termParamss.size == 1 then termParamss(0).params(0)
-        else termParamss(1).params(0)
-      }
+      if sym.isExtensionMethod then
+        sym.extendedTermParamLists.find(param => !param.isImplicit && !param.isGiven).flatMap(_.params.headOption)
+      else None
+
+    def splitExtensionParamList: (List[reflect.ParamClause], List[reflect.ParamClause]) =
+      import reflect.*
+      val method = sym.tree.asInstanceOf[DefDef]
```

Comment:
This cast is unchecked. You can make it checked using
```suggestion
      val method @ DefDef(_, _, _, _) = sym.tree
```

or 
```suggestion
      val method = sym.tree match
        case tree: DefDef => tree
```

Reply:
It's the assertion failure just as you said.